src/briefcase/integrations/docker.py

```diff
@@ -388,6 +399,18 @@ def check_output(self, args: SubprocessArgsT, image_tag: str, **kwargs) -> str:
             **self.dockerize_args(args, image_tag=image_tag, **kwargs)
         )
 
+    @classmethod
+    def subprocess_env(cls, env: dict[str, str] | None = None) -> dict[str, str]:
+        """Adds environment variables to the context that Docker runs in."""
+        env = {} if env is None else env
+        env.update(
+            {
+                # Disable the hints/recommendations that Docker prints in the console
+                "DOCKER_CLI_HINTS": "false",
+            }
+        )
```

Comment:
Two minor nits here: 

1. If the user provides an `env` dictionary as an argument, that dictionary will be modified in place - so using an environment in a subprocess will alter the environment
2. If, for some reason, you wanted to *enable* Docker hints (I don't know why, but work with me here 😄 ), there's no way to do so.

What about reversing the direction here: starting with a dictionary that contains `DOCKER_CLI_HINTS: "False"``, and updating *that* dictionary with env. That will ensure that the final dictionary is a copy, and any explicitly user-provided value will override the default.

Reply:
I should have realized this wouldve updated the dictionary input...which is definitely undesired. I updated it so this (and future) default can be overridden.